src/Shouldly/ShouldlyExtensionMethods/ShouldBe/ObjectGraphTestExtensions.cs

```diff
@@ -147,11 +147,19 @@ public static partial class ObjectGraphTestExtensions
         {
             foreach (var property in properties)
             {
-                var actualValue = property.GetValue(actual, Array.Empty<object>());
-                var expectedValue = property.GetValue(expected, Array.Empty<object>());
-
-                var newPath = path.Concat(new[] { property.Name });
-                CompareObjects(actualValue, expectedValue, newPath.ToList(), previousComparisons, customMessage, shouldlyMethod);
+                if (property.GetIndexParameters().Length == 0)
+                {
+                    var actualValue = property.GetValue(actual, Array.Empty<object>());
+                    var expectedValue = property.GetValue(expected, Array.Empty<object>());
+                    var newPath = path.Concat(new[] { property.Name });
+                    CompareObjects(actualValue, expectedValue, newPath.ToList(), previousComparisons, customMessage, shouldlyMethod);
+                }
+                else
+                {
+                    // There's no sensible way to compare indexers, as there does not exist a way to obtain a collection
+                    // of all values in a way that's common to all indexer implementations.
+                    throw new ShouldAssertException("Comparing indexer property types is not supported");
```

Comment:
I have a slight preference for keeping related things together, and I think of the GetIndexParameters check and this `throw` as a logical unit. If you don't feel strongly, can you invert the `if` and remove the `else` nesting?